enterprise/frontend/src/embedding-sdk/hooks/private/use-init-data.ts

```diff
@@ -49,22 +68,72 @@ export const useInitData = ({ config }: InitDataLoaderParameters) => {
   }, [dispatch, config.fetchRequestToken]);
 
   useEffect(() => {
-    if (loginStatus.status === "uninitialized") {
-      api.basename = config.metabaseInstanceUrl;
-
-      if (isValidJwtAuth(config)) {
-        setupJwtAuth(config, dispatch);
-        dispatch(setLoginStatus({ status: "validated" }));
-      } else {
-        dispatch(
-          setLoginStatus({
-            status: "error",
-            error: new Error(t`Invalid JWT URI provided.`),
-          }),
-        );
-      }
+    if (loginStatus.status !== "uninitialized") {
+      return;
+    }
+
+    api.basename = config.metabaseInstanceUrl;
+
+    const authErrorMessage = match<[SDKConfig, string], string | void>([
+      config,
+      window.location.pathname,
+    ])
+      .with(
+        [
+          {
+            apiKey: P.select("apiKey"),
+            jwtProviderUri: P.select("jwtProviderUri", P.nullish),
+          },
+          "localhost",
+        ],
+        ({ apiKey }) => {
+          presentApiKeyUsageWarning(appName);
+          setupLocalApiKey(dispatch, apiKey);
+        },
+      )
+      .with(
+        [
+          {
+            apiKey: P.select("apiKey"),
+            jwtProviderUri: P.select("jwtProviderUri", P.nullish),
+          },
```

Comment:
This is equivalent to
```suggestion
          {
            jwtProviderUri: P.nullish,
          },
```

since you don't specify the type for the `apiKey` and also didn't use it in the object handler.